lib/fsm/utils.go

```diff
@@ -139,6 +140,64 @@ func ResolvePlan(plan storage.OperationPlan, changelog storage.PlanChangelog) *s
 	return &plan
 }
 
+// DiffPlan returns the difference between the previous and the next plans in the
+// form of a changelog.
+func DiffPlan(prevPlan *storage.OperationPlan, nextPlan storage.OperationPlan) (diff []storage.PlanChange, err error) {
```

Comment:
nit: pass both plans as pointers. Or is there a reason for passing nextPlan as a struct?

Reply:
Yeah, it's actually on purpose. To indicate that the `prevPlan` is optional (nil can be passed), while the `nextPlan` should always be passed in.